aten/src/ATen/native/mps/operations/RangeFactories.mm

```diff
@@ -53,8 +53,57 @@
     }
     bool is_contiguous = result.is_contiguous();
     Tensor r = !is_contiguous ? at::empty_like(result, LEGACY_CONTIGUOUS_MEMORY_FORMAT) : result;
+    using namespace mps;
+    struct CachedGraph : public MPSCachedGraph {
+      CachedGraph(MPSGraph *graph) : MPSCachedGraph(graph) {}
+      MPSGraphTensor *startTensor = nil;
+      MPSGraphTensor *multiplyTensor = nil;
+      MPSGraphTensor *outputTensor = nil;
+    };
+    auto cache_ = MPSGraphCache::getInstance();
+    auto stream = getCurrentMPSStream();
+    auto mpsDataType = getMPSDataType(result.scalar_type());
+    @autoreleasepool {
+      string key = "arange_mps_out:" + getTensorsStringKey({result}) + ":" + to_string(size);
+      auto cachedGraph = static_cast<CachedGraph *>(cache_->LookUp(key));
+      if (!cachedGraph) {
+        auto *tmpCachedGraph = cache_->CreateCachedGraph(key, ^ MPSCachedGraph *() {
+          CachedGraph *newCachedGraph = nil;
+          @autoreleasepool {
+            auto mpsGraph = make_mps_graph();
+            newCachedGraph = new CachedGraph(mpsGraph);
+            auto shapeTensor = [mpsGraph constantWithData:[NSData dataWithBytes:&size length:sizeof(size)]
+                                                    shape: @[@1]
+                                                 dataType:MPSDataTypeInt64];
```

Comment:
@malfet I think we need to use Int32 type here with coordinateAlongAxis.

Reply:
Why? There are tests (for both int64 and float32 datatypes, which work as expected